src/xenia/cpu/hir/value.cc

```diff
@@ -77,8 +77,47 @@ uint64_t Value::AsUint64() {
 }
 
 void Value::Cast(TypeName target_type) {
-  // TODO(benvanik): big matrix.
-  assert_always();
+  switch (type) {
+    case INT32_TYPE:
+      switch (target_type) {
+        case FLOAT32_TYPE:
+          // ???
+          constant.f32 = constant.f32;
```

Comment:
No question marks needed - or the assignment - as a constant op this is fine just being the type change